content/redirect.js

```diff
@@ -2,18 +2,68 @@ const fs = require("fs");
 const path = require("path");
 
 const { resolveFundamental } = require("../libs/fundamental-redirects");
-const { CONTENT_ROOT, VALID_LOCALES } = require("./constants");
+const { decodePath, slugToFolder } = require("../libs/slug-utils");
+const {
+  CONTENT_ROOT,
+  CONTENT_TRANSLATED_ROOT,
+  VALID_LOCALES,
+} = require("./constants");
+
+const FORBIDDEN_URL_SYMBOLS = ["\n", "\t"];
+
+let ARCHIVED_URLS;
+
+function checkURLInvalidSymbols(url) {
+  for (const character of FORBIDDEN_URL_SYMBOLS) {
+    if (url.includes(character)) {
+      throw new Error(`URL contains invalid character '${character}'`);
+    }
+  }
+}
+
+function documentExists(url) {
```

Comment:
Nit: What does this function do? It sounds like it's asking "Does a document exist by this URL?" but it doesn't return `true` or `false`. It returns a string or a `null`.

Reply:
Perhaps something like `getPathFromURL(url)` then I think it makes sense to return a string or `null`.